infrastructure/src/main/java/org/corfudb/infrastructure/management/FileSystemAdvisor.java

```diff
@@ -0,0 +1,73 @@
+package org.corfudb.infrastructure.management;
+
+import lombok.extern.slf4j.Slf4j;
+import org.corfudb.protocols.wireprotocol.ClusterState;
+import org.corfudb.protocols.wireprotocol.NodeState;
+import org.corfudb.protocols.wireprotocol.failuredetector.FileSystemStats;
+import org.corfudb.protocols.wireprotocol.failuredetector.NodeRank.NodeRankByResourceQuota;
+
+import java.util.NavigableSet;
+import java.util.Optional;
+import java.util.TreeSet;
+
+/**
+ * Detects failures in the file system
+ */
+@Slf4j
+public class FileSystemAdvisor {
+
+    /**
```

Comment:
You should try to avoid static methods as much as possible, this class is used in other classes and could be replaced with something else in tests, if we make methods static it makes it much less flexible. 
So, just because this class doesn't have any fields doesn't mean it needs to be static